iidm/iidm-impl/src/test/java/com/powsybl/iidm/network/impl/NetworksTest.java

```diff
@@ -25,32 +23,28 @@
 public class NetworksTest {
 
     @Test
-    public void printBalanceSummaryTest()  {
-        StringBuilder buffer = new StringBuilder();
-
-        Logger logger = Mockito.mock(Logger.class);
-        Mockito.when(logger.isDebugEnabled()).thenReturn(true);
-        Mockito.doAnswer(invocationOnMock -> {
-            Object[] args = invocationOnMock.getArguments();
-            FormattingTuple formatter = MessageFormatter.format(Objects.toString(args[0]), args[1], args[2]);
-            buffer.append(formatter.getMessage());
-            return null;
-        }).when(logger).debug(Mockito.anyString(), Mockito.any(), Mockito.any());
-
-        Network network = EurostagTutorialExample1Factory.create();
-        Networks.printBalanceSummary("", network, logger);
-        assertEquals("Active balance at step '':\n" +
-                     "+-----------------------+--------------------------------+----------------------------------+\n" +
-                     "|                       | Main CC connected/disconnected | Others CC connected/disconnected |\n" +
-                     "+-----------------------+--------------------------------+----------------------------------+\n" +
-                     "| Bus count             |               4                |                0                 |\n" +
-                     "| Load count            | 1           | 0                | 0           | 0                  |\n" +
-                     "| Load (MW)             | 600.0       | 0.0              | 0.0         | 0.0                |\n" +
-                     "| Generator count       | 1           | 0                | 0           | 0                  |\n" +
-                     "| Max generation (MW)   | 9999.99     | 0.0              | 0.0         | 0.0                |\n" +
-                     "| Generation (MW)       | 607.0       | 0.0              | 0.0         | 0.0                |\n" +
-                     "| Shunt at nom V (MVar) | 0.0 0.0 (0) | 0.0 0.0 (0)      | 0.0 0.0 (0) | 0.0 0.0 (0)        |\n" +
-                     "+-----------------------+-------------+------------------+-------------+--------------------+" + System.lineSeparator(),
-                buffer.toString());
+    public void printBalanceSummaryTest() throws IOException {
+        try (Writer writer = new StringWriter()) {
+            Network network = EurostagTutorialExample1Factory.createWithMultipleConnectedComponents();
+            Networks.printBalanceSummary("", network, writer);
+            assertEquals("Active balance at step '':" + System.lineSeparator() +
+                            "+-----------------------+--------------------------------+----------------------------------+" + System.lineSeparator() +
```

Comment:
Be careful, the API we use to render ASCII tables doesn’t handle properly the line separator. It uses \n even on windows. It explains why the compilation fails